homeassistant/components/incomfort/sensor.py

```diff
@@ -0,0 +1,112 @@
+"""Support for an Intergas boiler via an InComfort/InTouch Lan2RF gateway."""
+from homeassistant.const import (
+    PRESSURE_BAR, TEMP_CELSIUS,
+    DEVICE_CLASS_PRESSURE, DEVICE_CLASS_TEMPERATURE)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from . import DOMAIN
+
+INTOUCH_HEATER_TEMP = 'CV Temp'
+INTOUCH_PRESSURE = 'CV Pressure'
+INTOUCH_TAP_TEMP = 'Tap Temp'
+
+INTOUCH_MAP_ATTRS = {
+    INTOUCH_HEATER_TEMP: ['heater_temp', 'is_pumping'],
+    INTOUCH_TAP_TEMP: ['tap_temp', 'is_tapping'],
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up an InComfort/InTouch sensor device."""
+    client = hass.data[DOMAIN]['client']
+    heater = hass.data[DOMAIN]['heater']
+
+    async_add_entities([
+        IncomfortPressure(client, heater),
+        IncomfortTemperature(client, heater, INTOUCH_HEATER_TEMP),
+        IncomfortTemperature(client, heater, INTOUCH_TAP_TEMP)
+    ])
+
+
+class IncomfortSensor(Entity):
+    """Representation of an InComfort/InTouch sensor device."""
+
+    def __init__(self, client, boiler):
+        """Initialize the sensor."""
+        self._client = client
+        self._boiler = boiler
+
+        self._name = None
+        self._device_class = None
+        self._unit_of_measurement = None
+
+    async def async_added_to_hass(self):
+        """Set up a listener when this entity is added to HA."""
+        async_dispatcher_connect(self.hass, DOMAIN, self._refresh)
+
+    @callback
+    def _refresh(self):
+        self.async_schedule_update_ha_state(force_refresh=True)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return self._device_class
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return self._unit_of_measurement
+
+    @property
+    def should_poll(self) -> bool:
+        """Return False as this device should never be polled."""
+        return False
+
+
+class IncomfortPressure(IncomfortSensor):
+    """Representation of an InTouch CV Pressure sensor."""
+
+    def __init__(self, client, boiler):
+        """Initialize the sensor."""
+        super().__init__(client, boiler)
+
+        self._name = INTOUCH_PRESSURE
+        self._device_class = DEVICE_CLASS_PRESSURE
+        self._unit_of_measurement = PRESSURE_BAR
```

Comment:
This is not allowed. Device class `pressure` is standardized and you need to use either `hPa` or `mbar` as per the docs https://developers.home-assistant.io/docs/en/entity_sensor.html#available-device-classes

Reply:
@Balloob 

This is not *air (gas) pressure*, it is *water (liquid) pressure* (of the circulating volume in the central heating system).

The entire EU reports water pressure is in **bar** (although natural gas pressure is in **mbar**) - I understand it is **psi** in the US.

In this case, the vendor very definitely reports the pressure in **bar**, and the value is between 0.0 and 3.x (usual range 1.0-2.0), with a precision of 0.1 - I can supply a reference to the vendor's documentation if required.

Also, I note that in **const.py** you have both X and mX for several other units, e.g.:
https://github.com/home-assistant/home-assistant/blob/2e848c3f1f5ecb4ed1127c1553aa89e2bc8e2d42/homeassistant/const.py#L353-L355

This makes sense, you wouldn't for example force people to use only **m** for distance when **km** (or vise-versa) makes more sense.

I (did read but) didn't pay too much attention to https://developers.home-assistant.io/docs/en/entity_sensor.html#available-device-classes because there appears to be a disconnect from it and the constants in `const.py` - I just assumed it was an anachronism:
 - `POWER_KILO_WATT = 'kW'` is in docs, but missing in **const.py**
 - `PRESSURE_INHG = 'inHg'` is in **const.py**, but missing from docs

Reply:
PRs for integrations are not the place to change the definition of a device class.

Reply:
So, am I understanding that I must first submit a separate PR to add **bar** as an option for the Pressure Sensor, and that it would get merged?  Am I missing something, and it's more complicated than that?

Or can I just do what **ebusd** and **opentherm_gw** have done and just have
```python
self._unit_of_measurement = 'bar'
```

Or are you simply saying it has to be **mbar** and that's that?

Reply:
Older code or code that got passed the code review while the rules were already in place is never a reason to add more tech debt. And so we won't do that.

Remove the device class and you can use any unit of measurement that you want.

Reply:
OK, I got it now. `DEVICE_CLASS_PRESSURE` has been removed, thanks.

I left in `PRESSURE_BAR`, but can can leave it if you require.